pkg/agent/plugin/nodeattestor/aws/iid.go

```diff
@@ -105,12 +153,15 @@ func (p *IIDAttestorPlugin) Configure(ctx context.Context, req *spi.ConfigureReq
 	// Set local vars from config struct
 	config.trustDomain = req.GlobalConfig.TrustDomain
 
-	if config.IdentityDocumentURL == "" {
-		config.IdentityDocumentURL = defaultIdentityDocumentURL
-	}
+	// If we have a legacy config, ensure both have a value
+	if config.EC2MetadataEndpoint == "" && (config.IdentitySignatureURL != "" || config.IdentityDocumentURL != "") {
```

Comment:
We should log here if deprecated configurables are set and the new endpoint configurable is:
1) set  (e.g. "deprecated configurable foo is ignored because...")
2) not set  (.e.g. "deprecated configurable foo will be deprecated in a future release. please use ec2_metadata_endpoint instead")